src/Visualization/Python/PlotPowerMonitors.py

```diff
@@ -0,0 +1,323 @@
+    for name in block_or_group_names:
+        if not (name in all_block_groups or name in all_block_names):
+            raise click.UsageError(
+                f"'{name}' matches no block or group name. "
+                f"Available names are: {all_block_groups + all_block_names}")
+
+    # Print available variables and exit
+    all_vars = volfiles[0].list_tensor_components(obs_id)
+    if list_vars or not vars_patterns:
+        import rich.columns
+        rich.print(rich.columns.Columns(all_vars))
+        return
+    # Expand globs in vars
+    vars = []
+    for var_pattern in vars_patterns:
+        matched_vars = fnmatch.filter(all_vars, var_pattern)
+        if not matched_vars:
+            raise click.UsageError(
+                f"The pattern '{var_pattern}' matches no variables. "
+                f"Available variables are: {all_vars}")
+        vars.extend(matched_vars)
+    # Remove duplicates. Ordering is lost, but that's not important here.
+    vars = list(set(vars))
+
+    # Apply stylesheets
+    stylesheets = [os.path.join(os.path.dirname(__file__), "plots.mplstyle")]
+    if stylesheet is not None:
+        stylesheets.append(stylesheet)
+    plt.style.use(stylesheets)
+
+    # Plot!
+    import rich.progress
+    try:
+        progress_cols = (
+            rich.progress.TextColumn(
+                "[progress.description]{task.description}"),
+            rich.progress.BarColumn(),
+            # Added in rich v12.0
+            rich.progress.MofNCompleteColumn(),
+            rich.progress.TimeRemainingColumn(),
+        )
+    except AttributeError:
+        progress_cols = (
+            rich.progress.TextColumn(
+                "[progress.description]{task.description}"),
+            rich.progress.BarColumn(),
+            rich.progress.TextColumn(
+                "[progress.percentage]{task.percentage:>3.0f}%"),
+            rich.progress.TimeRemainingColumn(),
+        )
+    progress = rich.progress.Progress(*progress_cols,
+                                      disable=(len(volfiles) == 1))
+    task_id = progress.add_task("Processing files")
+    volfiles_progress = progress.track(volfiles, task_id=task_id)
+    with progress:
+        plot_power_monitors(volfiles_progress,
+                            obs_id=obs_id,
+                            tensor_components=vars,
+                            domain=domain,
+                            block_or_group_names=block_or_group_names)
+        progress.update(task_id, completed=len(volfiles))
+
+    if output:
+        plt.savefig(output)
+    else:
+        if not os.environ.get("DISPLAY"):
+            logger.warning(
+                "No 'DISPLAY' environment variable is configured so plotting "
+                "interactively is unlikely to work. Write the plot to a file "
+                "with the --output/-o option.")
+        plt.show()
+
+
+if __name__ == "__main__":
+    plot_power_monitors_command(help_option_names=["-h", "--help"])
```

Comment:
When I run this with `-h` what general info do I get? I find it useful to have a description of what the code does in addition to the options.

Reply:
Thanks for catching this, I forgot to add a docstring here. See fixup.